react-components/src/higher-order-components/withCameraStateUrlParam.tsx

```diff
@@ -0,0 +1,86 @@
+/*!
+ * Copyright 2023 Cognite AS
+ */
+
+import { useEffect, type ReactElement, type FunctionComponent } from 'react';
+import { useReveal } from '..';
+import { Vector3 } from 'three';
+import { type CameraState } from '@cognite/reveal';
+
+type CameraStateTransform = Omit<Required<CameraState>, 'rotation'>;
+
+export function withCameraStateUrlParam<T extends object>(
+  Component: FunctionComponent<T>
+): FunctionComponent<T> {
+  return function CameraStateUrlParam(props: T): ReactElement {
+    const reveal = useReveal();
+    const getCameraStateFromUrlParam = useGetCameraStateFromUrlParam();
+
+    const setUrlParamOnCameraStop = (): void => {
+      const currentUrlCameraState = getCameraStateFromUrlParam();
+      const currentCameraManagerState = reveal.cameraManager.getCameraState();
+      if (
+        currentUrlCameraState !== undefined &&
+        !hasCameraStateChanged(currentUrlCameraState, currentCameraManagerState)
+      ) {
+        return;
+      }
+      const { position, target } = currentCameraManagerState;
+      const url = new URL(window.location.toString());
+      url.searchParams.set('cameraPosition', `[${position.x},${position.y},${position.z}]`);
+      url.searchParams.set('cameraTarget', `[${target.x},${target.y},${target.z}]`);
+      window.history.pushState({}, '', url);
+    };
+
+    useEffect(() => {
+      reveal.cameraManager.on('cameraStop', setUrlParamOnCameraStop);
+      return () => {
+        reveal.cameraManager.off('cameraStop', setUrlParamOnCameraStop);
+      };
+    }, []);
+
+    return <Component {...props} />;
+  };
+
+  function hasCameraStateChanged(
+    previous: CameraStateTransform,
+    current: CameraStateTransform
+  ): boolean {
+    const epsilon = 0.001;
```

Comment:
Disagree, first of all I don't see the point in refactoring it out into a separate file since it is only used once. We can consider if it is used multiple times in different places. Having a utilities/constants.ts in the first place seems like a mistake to me. Should every constant in the entire code base be here? How does that make anything easier? In the very least it should be grouped by function. I don't f.ex. see what `DEFAULT_QUERY_STALE_TIME` has to do with `METERS_TO_FEET`; they are not related what so ever.